src/Illuminate/Database/Query/Builder.php

```diff
@@ -1427,31 +1441,33 @@ public function simplePaginate($perPage = 15, $columns = ['*'], $pageName = 'pag
      */
     public function getCountForPagination($columns = ['*'])
     {
-        $this->backupFieldsForCount();
+        $selectSql = $this->toSql();
```

Comment:
Yeh, I know that does that. I want to keep that behaviour, but not use that style.